diff --git a/arch/arm/boot/dts/imx6q-iris2-R0.dts b/arch/arm/boot/dts/imx6q-iris2-R0.dts
index 821b6d668a12..2b5be9c13d22 100644
--- a/arch/arm/boot/dts/imx6q-iris2-R0.dts
+++ b/arch/arm/boot/dts/imx6q-iris2-R0.dts
@@ -163,6 +163,7 @@
 &uart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart2_1>;
+	dma-names = "", "";
 	status = "okay";
 };
 
@@ -177,6 +178,7 @@
 &uart4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart4_1>;
+	dma-names = "", "";
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6q-iris2-R1.dts b/arch/arm/boot/dts/imx6q-iris2-R1.dts
index 578b3196b9ef..243906f13715 100644
--- a/arch/arm/boot/dts/imx6q-iris2-R1.dts
+++ b/arch/arm/boot/dts/imx6q-iris2-R1.dts
@@ -238,6 +238,7 @@
 &uart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart2_1>;
+	dma-names = "", "";
 	status = "okay";
 };
 
@@ -252,6 +253,7 @@
 &uart4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart4_1>;
+	dma-names = "", "";
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6q-nightcrawler-R0.dts b/arch/arm/boot/dts/imx6q-nightcrawler-R0.dts
index 40cae1f40259..dae810bb862e 100644
--- a/arch/arm/boot/dts/imx6q-nightcrawler-R0.dts
+++ b/arch/arm/boot/dts/imx6q-nightcrawler-R0.dts
@@ -230,6 +230,7 @@
 &uart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart2_3>;
+	dma-names = "", "";
 	status = "okay";
 };
 
@@ -244,6 +245,7 @@
 &uart4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart4_1>;
+	dma-names = "", "";
 	status = "okay";
 };
 
