diff --git a/nvidia/platform/t19x/jakku/kernel-dts/common/tegra194-p3509-disp.dtsi b/nvidia/platform/t19x/jakku/kernel-dts/common/tegra194-p3509-disp.dtsi
index 3a09853c4983..f2459ad1e0c7 100644
--- a/nvidia/platform/t19x/jakku/kernel-dts/common/tegra194-p3509-disp.dtsi
+++ b/nvidia/platform/t19x/jakku/kernel-dts/common/tegra194-p3509-disp.dtsi
@@ -27,7 +27,7 @@
 #include "tegra194-spmic-p3668.dtsi"
 
 &head0 {
-	status = "okay";
+	status = "disabled";
 	nvidia,fb-bpp = <32>;
 	nvidia,fbmem-size = <265420800>; /* 8K (7680*4320) 32bpp double buffered */
 	nvidia,fb-flags = <TEGRA_FB_FLIP_ON_PROBE>;
@@ -41,7 +41,7 @@
 };
 
 &head1 {
-	status = "okay";
+	status = "disasbled";
 	nvidia,fb-bpp = <32>;
 	nvidia,fbmem-size = <265420800>; /* 8K (7680*4320) 32bpp double buffered */
 	nvidia,fb-flags = <TEGRA_FB_FLIP_ON_PROBE>;
@@ -57,35 +57,35 @@
 };
 
 &sor0 {
-	status = "okay";
+	status = "disabled";
 	nvidia,active-panel = <&sor0_dp_display>;
 };
 
 &sor0_dp_display {
-	status = "okay";
+	status = "disabled";
 	nvidia,is_ext_dp_panel = <1>;
 };
 
 &sor1 {
-	status = "okay";
+	status = "disabled";
 	nvidia,active-panel = <&sor1_hdmi_display>;
 };
 
 &sor1_hdmi_display {
-	status = "okay";
+	status = "disabled";
 	disp-default-out {
 		nvidia,out-flags = <TEGRA_DC_OUT_HOTPLUG_LOW>;
 	};
 };
 
 &dpaux0 {
-	status = "okay";
+	status = "disabled";
 };
 
 &dpaux1 {
-	status = "okay";
+	status = "disabled";
 };
 
 &tegra_cec {
-	status = "okay";
+	status = "disabled";
 };
diff --git a/nvidia/platform/t19x/jakku/kernel-dts/common/tegra194-p3668-common.dtsi b/nvidia/platform/t19x/jakku/kernel-dts/common/tegra194-p3668-common.dtsi
index ba0fff2c8052..5c08a86d72e3 100644
--- a/nvidia/platform/t19x/jakku/kernel-dts/common/tegra194-p3668-common.dtsi
+++ b/nvidia/platform/t19x/jakku/kernel-dts/common/tegra194-p3668-common.dtsi
@@ -209,7 +209,7 @@
 
 	host1x {
 		dpaux@155F0000 {
-			status = "okay";
+			status = "disabled";
 			compatible = "nvidia,tegra194-dpaux3-padctl";
 			/delete-property/ power-domains;
 			dpaux_default: pinmux@0 {
