diff --git a/nvidia/platform/t210/porg/kernel-dts/tegra210-porg-p3448-common.dtsi b/nvidia/platform/t210/porg/kernel-dts/tegra210-porg-p3448-common.dtsi
index 2a7d84445089..3cfcf5f7a845 100644
--- a/nvidia/platform/t210/porg/kernel-dts/tegra210-porg-p3448-common.dtsi
+++ b/nvidia/platform/t210/porg/kernel-dts/tegra210-porg-p3448-common.dtsi
@@ -111,7 +111,7 @@
 
 		/* tegradc.0 */
 		dc@54200000 {
-			status = "okay";
+			status = "diabled";
 			nvidia,dc-flags = <TEGRA_DC_FLAG_ENABLED>;
 			nvidia,emc-clk-rate = <300000000>;
 			nvidia,cmu-enable = <1>;
@@ -122,7 +122,7 @@
 			win-mask = <0x7>; /* Assign only wins A/B/C */
 		};
 		dc@54240000 {
-			status = "okay";
+			status = "disabled";
 			nvidia,dc-flags = <TEGRA_DC_FLAG_ENABLED>;
 			nvidia,emc-clk-rate = <300000000>;
 			nvidia,cmu-enable = <1>;
@@ -133,10 +133,10 @@
 			win-mask = <0x7>; /* Assign only wins A/B/C */
 		};
 		sor {
-			status = "okay";
+			status = "disabled";
 			nvidia,xbar-ctrl = <2 1 0 3 4>;
 			dp-display {
-				status = "okay";
+				status = "disabled";
 			};
 		};
 		sor1 {
@@ -146,16 +146,16 @@
 			 * SOR1 and the pad.
 			 */
 			nvidia,xbar-ctrl = <0 1 2 3 4>;
-			status = "okay";
+			status = "disabled";
 			hdmi-display {
-				status = "okay";
+				status = "disabled";
 			};
 		};
 		dpaux {
-			status = "okay";
+			status = "disabled";
 		};
 		dpaux1 {
-			status = "okay";
+			status = "disabled";
 		};
 	};
 
