diff --git a/H264_UVC_TestAP.c b/H264_UVC_TestAP.c
index 7a86fbe..a5ddb6c 100644
--- a/H264_UVC_TestAP.c
+++ b/H264_UVC_TestAP.c
@@ -143,8 +143,8 @@ static int CheckKernelVersion(void)
 	else
 	{
 		TestAp_Printf(TESTAP_DBG_ERR, "your kernel version: 0x%x \nTestAP support kernel version: 0x%x\n",kernelRelease, LINUX_VERSION_CODE);
-		return false;
 	}
+	return true;
 
 }
 static int GetFreeRam(int* freeram)
@@ -1858,6 +1858,7 @@ int main(int argc, char *argv[])
 			break;
 
 		case OPT_H264_IFRAME_SET:
+			TestAp_Printf(TESTAP_DBG_ERR, "IFrame chosen\n");
 			h264_iframe_reset = atoi(optarg);
 			do_h264_iframe_set = 1;
 			break;
@@ -2121,6 +2122,11 @@ int main(int argc, char *argv[])
 			TestAp_Printf(TESTAP_DBG_ERR, "RERVISION_UVC_TestAP @main : XU_H264_Set_BitRate Failed\n");
 	}
 
+	if(do_h264_iframe_set)
+	{
+		XU_H264_Set_IFRAME(dev);
+	}
+
 	if(do_xu_set)
 	{
 		TestAp_Printf(TESTAP_DBG_FLOW, "== XU Set: input command ==\n");		
